src/java.base/share/classes/java/lang/String.java

```diff
@@ -5301,6 +5310,7 @@ void getBytes(byte[] dst, int srcPos, int dstBegin, byte coder, int length) {
                     this.value = buf;
                     return;
                 }
+                initCompressionFlag();
```

Comment:
This call should not be conditional on `asb.maybeLatin1`.

Reply:
Changed this so it's in its own `if(COMPACT_STRINGS)` check.